File: BFinanceBackend/BFinanceModels/Subscription.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace BFinanceModels
{
    public partial class Subscription
    {
        public int SubscriptionId { get; set; }
        public string SubscriptionName { get; set; }
        public decimal PaymentAmount { get; set; }
        public string PaymentPeriod { get; set; }
        public string PaymentDueDate { get; set; }
        public int FromAccountId { get; set; }

        public virtual Account FromAccount { get; set; }
    }
}

File: BFinanceFrontend/src/app/Models/subscription.ts
import { GetAccount } from "./account";

export interface GetSubscription {
  subscriptionId: number;
  subscriptionName: string;
  paymentAmount: number;
  paymentPeriod: string;
  paymentDueDate: string;
  fromAccountId: number;
  fromAccount: GetAccount;
}

export interface PostSubscription {
  subscriptionName: string;
  paymentAmount: number;
  paymentPeriod: string;
  paymentDueDate: string;
  fromAccountId: number;
}
